BLSQ/iaso
# Polio-1124 add lqas refresh button

LQAs country map needed a button to refresh data from country. This implies launching an OpenHexa pipeline

Related JIRA tickets :  POLIO-1124

## Self proofreading checklist


## Changes

- Add `RefreshLqas` component
    -  The component checks in the tasks lists the latest task for the country and displays its datetime info+status. It keeps polling the API every second to update the status and inform the user if a manual update is ongoing
    - If a manual update is ongoing, the user will not be able to launch a new task until the manual update is over
    - Else it will create a new refresh task and disable the update button until the task is done
- Add `taskMonitor.ts` with hooks to create tasks and poll `tasks` api for task update
- Add `useGetLatestLQASIMUpdate` to show latest manual update and its status
- Rearranged the LQAs Afro map filters to have a better layout on small screens. No visible difference otherwise

## How to test

### Pre-requisites
- Admin access to the [OpenHexa workspace](https://app.openhexa.org/workspaces/polio-3688d8)
- An `lqas-config` json config file to get lqas data 
- ngrok

### Setup
- Run `docker-compose up` or `docker-compose up iaso`
- Start an ngrok tunnel on localhost:8081
- In OpenHexa: 
    - Change the `iaso-tester` connection with the ngrok url and your local Db credentials
    - Go to "Pipelines", click "Create" and copy your API token
- In your `.env` : 
    - Set `OPENHEXA_TOKEN` to the value just copied from OpenHexa 
    - Set `OH_PIPELINE_TARGET` to `"custom"`
    - Set `OPENHEXA_URL` to  `"https://api.openhexa.org/graphql/"`
    - Set `LQAS_PIPELINE_VERSION` to 74
    - Set `LQAS_PIPELINE`: see setup for https://github.com/BLSQ/iaso-pipelines/pull/15
- If you already have lqas data in your DB:
    - Go to django admin > Jsondatastore and pick an lqas datastore (naming convention `lqas_<country_id >`)
    - In the json, delete entries at the `data` level of the nicely nested object. This will allow you to see the changes when launching the update.

### Test
- Press the button, see what happens.

## Print screen / video


https://github.com/BLSQ/iaso/assets/38907762/91aaeb7d-1b6b-4120-a8f3-8cdee174f71f



## Notes

Merges in https://github.com/BLSQ/iaso/tree/POLIO-1191_add_endpoint_for_openhexa_task
Requires https://github.com/BLSQ/iaso-pipelines/pull/15

VERY IMPORTANT: environment variables need to be added to polio and polio staging. The `OPENHEXA_TOKEN`should be taken from the `OpenHexa API` connection in OpenHexa's polio workspace
